src/openrct2-ui/windows/Scenery.cpp

```diff
@@ -1186,6 +1202,11 @@ class SceneryWindow final : public Window
         if (scenery.IsUndefined())
             return;
 
+        auto lastScenery = GetSelectedScenery(_activeTabIndex);
+        if (!(lastScenery == scenery || IsFiltered(lastScenery)))
```

Comment:
Inverting this check makes it easier to understand IMO.
```suggestion
        if (lastScenery != scenery && !IsFiltered(lastScenery))
```

Also the name `IsFiltered` is not exactly clear. Does it mean the object passed the filter, or is it excluded by the filter? I'm not sure what a better name is without making it much longer.

Reply:
What about `MatchFilter`

Reply:
Also I think I did it that way because `ScenerySelection` does not have an operator override for !=. I can add that in if needed though.